randovania/game_connection/builder/dolphin_connector_builder.py

```diff
@@ -11,12 +12,62 @@
 
 
 class DolphinConnectorBuilder(PrimeConnectorBuilder):
+    dolphin_cmd: str
+    dolphin_comm: str
+    dolphin_child: Popen
+
+    def __init__(self, dolphin_cmd: str, dolphin_comm: str):
+        super().__init__()
+        self.dolphin_cmd = dolphin_cmd
+        self.dolphin_comm = dolphin_comm
+        self.dolphin_child = None
+
+    @property
+    def pretty_text(self) -> str:
+        if self.dolphin_comm:
+            if self.dolphin_cmd:
+                return f"{super().pretty_text}: run '{self.dolphin_cmd}', attach to '{self.dolphin_comm}'"
+            else:
+                return f"{super().pretty_text}: autodetect as '{self.dolphin_comm}'"
+        elif self.dolphin_cmd:
+            return f"{super().pretty_text}: run '{self.dolphin_cmd}', then autodetect"
+        else:
+            return f"{super().pretty_text}: autodetect using defaults"
+
     @property
     def connector_builder_choice(self) -> ConnectorBuilderChoice:
         return ConnectorBuilderChoice.DOLPHIN
 
+    def start_dolphin(self) -> None:
+        if not self.dolphin_cmd:
+            return
+
+        if self.dolphin_child is None:
+            self.logger.info(f"Dolphin not running, attempting to start it: {self.dolphin_cmd}")
+            self.dolphin_child = Popen(self.dolphin_cmd, shell=True)
+
+        elif self.dolphin_child.poll() is not None:
+            self.logger.info(f"Dolphin has exited, restarting it: {self.dolphin_cmd}")
+            self.dolphin_child = Popen(self.dolphin_cmd, shell=True)
+
+    def stop_dolphin(self) -> None:
+        if self.dolphin_child is not None:
+            self.logger.info("Shutting down child Dolphin process.")
+            # Send TERM twice, since the first one just opens a "are you sure
+            # you want to exit?" popup.
```

Comment:
small style nit, but i'd split the comment so so that the quote is all on one line